logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/call/usecase/ObserveAskCallFeedbackUseCase.kt

```diff
@@ -17,26 +17,26 @@
  */
 package com.wire.kalium.logic.feature.call.usecase
 
+import com.wire.kalium.logic.feature.user.ShouldAskCallFeedbackUseCaseResult
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.filterIsInstance
 import kotlinx.coroutines.flow.map
 
 /**
- * The useCase for observing when the ongoing call was ended because of degradation of conversation verification status (Proteus or MLS)
+ * Use case to observe if we should ask for feedback after the call has ended.
  */
 interface ObserveAskCallFeedbackUseCase {
     /**
-     * @return [Flow] that emits only when the call was ended because of degradation of conversation verification status (Proteus or MLS)
+     * @return [Flow] that emits [ShouldAskCallFeedbackUseCaseResult] when the call has ended and we should ask for feedback.
      */
-    suspend operator fun invoke(): Flow<Boolean>
+    suspend operator fun invoke(): Flow<ShouldAskCallFeedbackUseCaseResult>
 }
 
-@Suppress("FunctionNaming")
-internal fun ObserveAskCallFeedbackUseCase(
+internal fun observeAskCallFeedbackUseCase(
```

Comment:
question: does it need to be a function that creates anonymous object? Usually we just create 
```
class ObserveAskCallFeedbackUseCaseImpl(
    endCallListener: EndCallResultListener
) : ObserveAskCallFeedbackUseCase {
```

Reply:
To be fair, I'd even go the extra mile to make it look like a constructor when invoking. something like `internal fun ObserveAskCallFeedbackUseCase`. Just looks cleaner than having all `Impl` classes everywhere :D